src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompilationInfo.cs

```diff
@@ -1,14 +1,19 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Text.RegularExpressions
 {
     public class RegexCompilationInfo
     {
-        private string _pattern;
-        private string _name;
-        private string _nspace;
+        [NotNull]
```

Comment:
fields will be initialized via property setters which are always called by the ctors. The analyzers can't detect this.

Reply:
We handle this pattern elsewhere with:
```C#
private string _pattern = null!; // initialized by setter called from ctor
```